### PR TITLE
namespace interfaces with underscore

### LINKED ISSUE #44: Reconsider alertmanager interface name
`alertmanager` is not a general relation name, it is more the `interface` of the relation. We need to find a more general name and use it as standard name for relations like "we push alerts".

### PR DESCRIPTION
**Pull request by sed-i**
Resolves #44

### CODE REVIEW

File: metadata.yaml [hunk 1]
```diff
@@ -33,7 +33,7 @@ requires:
     grafana-source:
         interface: grafana_datasource
     alertmanager:
-        interface: alertmanager
+        interface: alertmanager_dispatch
```
Comment (balbirthomas) [inline metadata.yaml:36]: It will be very helpful if you can add a commit message with the rationale underlying this choice of name.

**Review by balbirthomas — APPROVED** [review]

File: metadata.yaml (hunk 1 above)
Comment (sed-i) [reply metadata.yaml:36]: You mean in addition to the discussion in the linked issue #44?

Comment (balbirthomas) [reply metadata.yaml:36]: If you prefer you can just link to the issue. But a few lines copied and pasted  from there would also be be nice if you think git commit history is a valuable resource for future developers.

Comment (sed-i) [reply metadata.yaml:36]: Done:
https://github.com/canonical/alertmanager-operator/pull/14/commits/29514ca066294ef62cfcb5b28f58240a6d3fecf6

**Review by mmanciop — APPROVED** [review]

**Review by mmanciop — APPROVED** [review]

**Review by rbarry82 — APPROVED** [review]

**Pull request merged**